addons/pylons/functions/fnc_showDialog.sqf

```diff
@@ -0,0 +1,137 @@
+/*
+* Author: 654wak654
+* Shows the aircraft loadout dialog for given aircraft.
+*
+* Arguments:
+* 0: Aircraft <OBJECT>
+*
+* Return Value:
+* None
+*
+* Example:
+* [vehicle ace_player] call ace_pylons_fnc_showDialog
+*
+* Public: Yes
+*/
+#include "script_component.hpp"
+
+params ["_aircraft"];
+
+if (!GVAR(enabled) || {!(typeOf _aircraft in GVAR(aircraftWithPylons))}) exitWith {};
+
+private _currentUser = _aircraft getVariable [QGVAR(currentUser), objNull];
+if (!isNull _currentUser) exitWith {
+    [format [localize LSTRING(InUse), name _currentUser], false, 5] call EFUNC(common,displayText);
+};
+_aircraft setVariable [QGVAR(currentUser), ace_player, true];
+GVAR(currentAircraftNamespace) setVariable [getPlayerUID ace_player, _aircraft, true];
+[_aircraft, "blockEngine", QUOTE(ADDON), true] call EFUNC(common,statusEffect_set);
+
+GVAR(currentAircraft) = _aircraft;
+
+createDialog QGVAR(DialogLoadout);
+private _display = DISPLAY(ID_DIALOG);
+_display displayAddEventHandler ["Unload", LINKFUNC(onButtonClose)];
+
+if (GVAR(rearmNewPylons)) then {
+    ctrlShow [ID_TEXT_BANNER, false];
+    ctrlShow [ID_PICTURE_REARM, false];
+} else {
+    ctrlSetText [ID_TEXT_BANNER, "    " + (ctrlText ID_TEXT_BANNER)]; // Spacing for the icon
+};
+
+private _config = configFile >> "CfgVehicles" >> typeOf _aircraft;
+private _pylonComponent = _config >> "Components" >> "TransportPylonsComponent";
+
+ctrlSetText [ID_PICTURE_AIRCRAFT, getText (_pylonComponent >> "uiPicture")];
+
+private _hasFRIES = [_config >> QEGVAR(fastroping,enabled), "number", 0] call CBA_fnc_getConfigEntry;
+if (["ace_fastroping"] call EFUNC(common,isModLoaded) && {_hasFRIES > 1}) then {
+    private _checkbox = _display displayCtrl ID_CHECKBOX_FRIES;
+    private _fries = _aircraft getVariable [QEGVAR(fastroping,FRIES), objNull];
+    _checkbox cbSetChecked (!isNull _fries);
+    _checkbox setVariable [QGVAR(originalState), !isNull _fries];
+} else {
+    ctrlShow [ID_CHECKBOX_FRIES, false];
+    ctrlShow [ID_TEXT_FRIES, false];
+};
+
+GVAR(comboBoxes) = [];
+{
+    private _combo = _display ctrlCreate ["RscCombo", -1];
+    private _picturePos = ctrlPosition (_display displayCtrl ID_BACKGROUND_PICTURE);
+    private _uiPos = getArray (_x >> "UIposition");
```

Comment:
```
Error in expression <trlSetPosition [
(_picturePos select 0) + (_uiPos select 0),
(_picturePos select>
  Error position: <+ (_uiPos select 0),
(_picturePos select>
  Error Generic error in expression
File z\ace\addons\pylons\functions\fnc_showDialog.sqf, line 51
```
Array could contain strings that need to be evaluated
Example from CUP AH1Z:

`configfile >> "CfgVehicles" >> "CUP_B_AH1Z_Dynamic_USMC" >> "Components" >> "TransportPylonsComponent" >> "pylons" >> "pylons1" >> "UIposition"`

UIposition[] = {"0.0.4",0.3};

Reply:
Why? 😦

Reply:
Both `call compile "0.0.4"` and `parseNumber "0.0.4"` return 0.